### ISSUE TITLE
Errors involving core libraries are difficult to debug

### ISSUE THREAD

**Issue by pasha-mf** [facebook/flow#900]
Suppose I made an error and now Flow reports that my code is not compatible with some core type definitions:

```
var elem: HTMLInputElement = document.createElement("button");

js/flowtest.js:29:30,61: call of method createElement
Function cannot be called on
[LIB] dom.js:235:5,17: Document
```

Now, sometimes the cause of the error may be obvious; but other times -- it won't, and you'll want to look at the actual definition related to the error. And there is no easy way of doing so: I can go online and read https://github.com/facebook/flow/blob/master/lib/dom.js, but those definitions are updated all the time, and my system is not using the most up-to-date version. I can also restart Flow server and it will tell me where the log files are being stored, and if I go there I'll see a dozen or so cryptic directories, each containing core libraries. It's not exactly clear which one is currently in use, but there is a workaround: just delete all of them, restart the server and then come back.

In other words, it's a big hassle.
What would be more convenient is to have an option in .flowconfig file which would instruct the flow server where to store those core libraries. For example I'd store them in my project's lib/flow/core/ and then will be able to easily check those definitions at any time.

PS: I have a strong feeling that definition of Promise.catch in core.js is incorrect, since it has a different signature than the second parameter in Promise.then.

**Comment by mroch** [comment]
from your example it looks like you're using `--strip-root`. it should give you the full path to dom.js if you don't use that.

**Comment by samwgoldman** [comment]
Ultimately, this error message is bad because `createElement` is heavily overloaded, overloaded functions are implemented as intersection types, and intersection types (like union types) do not provide very helpful error messages.

This is a very well known problem, but unfortunately the solution is not as well known. Thanks for providing yet another example of this so we can test possible solutions against real world use cases.

@pasha-mf Regarding the issue with `Promise.catch`—can you open a separate issue, so we don't lose track of it?

**Comment by samwgoldman** [comment]
IMO this error is now much improved. Today Flow outputs this:

```
Error: test.js:3
  3: var elem: HTMLInputElement = document.createElement("button");
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ HTMLButtonElement. This type is incompatible with
  3: var elem: HTMLInputElement = document.createElement("button");
               ^^^^^^^^^^^^^^^^ HTMLInputElement
```
